### PR TITLE
PLKSR & RealPLKSR support

### LINKED ISSUE #251: PLKSR and RealPLKSR support
Could you add support for PLKSR and RealPLKSR model

### PR DESCRIPTION
**Pull request by joeyballentine**
~CI is currently failing due to the safetensors tests. I'm not sure why exactly. From the error it appears it might be saving 3x models as 2x for some reason, but the logging i put in clearly was printing out that the scale was 3, so i'm confused how that would be.~

~I also still need to pick tags for this~

Closes #251

### CODE REVIEW

File: libs/spandrel/spandrel/architectures/PLKSR/__init__.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                "feats.1.channe_mixer.0.weight",
+                "feats.1.channel_mixer.0.weight",
+            ),
+        )
+
+    @override
+    def load(self, state_dict: StateDict) -> ImageModelDescriptor[_PLKSR]:
+        scale: int
+        in_channels = 3
+        out_channels = 3
```
Comment (RunDevelopment) [inline libs/spandrel/spandrel/architectures/PLKSR/__init__.py:41]: Both are hard-coded to 3 in the arch. There's no reason to declare a variable here. Just inline the 3 as `input_channels=3` when creating the `ImageModelDescriptor` and add a comment saying that it's hard-coded.

File: libs/spandrel/spandrel/architectures/PLKSR/__init__.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            )
+            and KeyCondition.has_any(
+                "feats.1.channe_mixer.0.weight",
+                "feats.1.channel_mixer.0.weight",
+            ),
+        )
+
+    @override
+    def load(self, state_dict: StateDict) -> ImageModelDescriptor[_PLKSR]:
+        scale: int
```
Comment (RunDevelopment) [inline libs/spandrel/spandrel/architectures/PLKSR/__init__.py:39]: This declaration is unnecessary.

File: libs/spandrel/spandrel/architectures/PLKSR/__init__.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            scale_shape = state_dict[upscale_key].shape[0]
+            scale = math.isqrt(scale_shape // out_channels)
+
+        if "feats.1.lk.conv.weight" in state_dict:
+            kernel_size = state_dict["feats.1.lk.conv.weight"].shape[2]
+            after_split = state_dict["feats.1.lk.conv.weight"].shape[0]
+            split_ratio = after_split / dim
+
+        if "feats.1.attn.f.0.weight" not in state_dict:
+            use_ea = False
```
Comment (RunDevelopment) [inline libs/spandrel/spandrel/architectures/PLKSR/__init__.py:73]:
```suggestion
        use_ea = "feats.1.attn.f.0.weight" in state_dict

```

File: libs/spandrel/spandrel/architectures/PLKSR/__init__.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+        # Yes, the normal version has this typo.
+        if "feats.1.channe_mixer.0.weight" in state_dict:
+            self._name = "PLKSR"
```
Comment (RunDevelopment) [inline libs/spandrel/spandrel/architectures/PLKSR/__init__.py:77]: TIL that `_variable_name` class fields aren't private but protected from the perspective of the type checker.

Yeah, this was not intended. `Architecture` instances are supposed to be immutable. So this is not allowed. I would suggest adding "Real" as a tag for `RealPLKSR`.

File: tests/test_PLKSR.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+
+def test_load():
+    assert_loads_correctly(
+        PLKSRArch(),
+        lambda: PLKSR(),
+        lambda: PLKSR(dim=32),
+        lambda: PLKSR(upscaling_factor=2),
+        lambda: PLKSR(n_blocks=6),
+        lambda: PLKSR(kernel_size=9),
```
Comment (RunDevelopment) [inline tests/test_PLKSR.py:23]: Those aren't enough tests. You have a lot of logic for detecting e.g. `ccm_type` and none if it is tested here. Every parameter should be tested, and if a parameter has only a few possible values, all possible values should be tested.

**Review by RunDevelopment — COMMENTED** [review]
Nice work! I actually wanted to add PLKSR today, so this saves me some time :)

File: libs/spandrel/spandrel/architectures/PLKSR/__init__.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            id="PLKSR",
+            detect=KeyCondition.has_all(
+                "feats.0.weight",
+                "feats.1.lk.conv.weight",
+                "feats.1.refine.weight",
+            )
+            and KeyCondition.has_any(
+                "feats.1.channe_mixer.0.weight",
+                "feats.1.channel_mixer.0.weight",
+            ),
```
[the remark this answers is not in the document]
Comment (joeyballentine) [reply libs/spandrel/spandrel/architectures/PLKSR/__init__.py:34]: how do i do multiple conditions then?

[the remark this answers is not in the document]
Comment (joeyballentine) [reply libs/spandrel/spandrel/architectures/PLKSR/__init__.py]: nvm i found an example

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**